### PR TITLE
fixing bug in determining synapse type in multinode #99 #minor

### PR DESCRIPTION
**Pull request by anilbey**
Solves the bug introduced in #99.

### CODE REVIEW

File: tests/test_synapse/test_synapse_factory.py [hunk 1]
```diff
@@ -74,12 +74,26 @@ def test_synlocation_to_segx(self):
...
+    syn_description[SynapseProperty.TYPE] = 150
+    syn_description[SynapseProperties.plasticity[0]] = 1.0  # Just one property set
+    assert SynapseFactory.determine_synapse_type(syn_description) == SynapseType.AMPANMDA
+
+    # Test when synapse is excitatory with all plasticity properties filled
+    for prop in SynapseProperties.plasticity:
+        syn_description[prop] = 1.0  # Assign a valid value to each plasticity property
+    assert SynapseFactory.determine_synapse_type(syn_description) == SynapseType.GLUSYNAPSE
+
+    # Test when synapse is excitatory with one plasticity property as NaN
```
Comment (AurelienJaquier) [inline tests/test_synapse/test_synapse_factory.py:97]: this is the same test as in line 90, no?

Comment (anilbey) [reply tests/test_synapse/test_synapse_factory.py:97]: Actually it tests a slightly different case
- On line 90 the values are [150, 1.0]
- On line 99 the values are [150, nan, 1.0, 1.0, 1.0, 1.0, 1.0]
Both should evaluate to AMPANMDA.

Comment (AurelienJaquier) [reply tests/test_synapse/test_synapse_factory.py:97]: I thought on line 90 it was [1.0, nan, 150]?
Either way, if you say they are different, it's ok.

**Review by AurelienJaquier — APPROVED** [review]

File: tests/test_synapse/test_synapse_factory.py (hunk 1 above)
Comment (anilbey) [reply tests/test_synapse/test_synapse_factory.py:97]: Let me put a debugger and double check the exact values.

Comment (anilbey) [reply tests/test_synapse/test_synapse_factory.py]: I updated the implementation to raise an exception when the synapse type cannot be determined.

Comment (anilbey) [reply tests/test_synapse/test_synapse_factory.py]: Before it was assuming the synapse type to be AMPANMDA when some plasticity params (not all) were missing. That could lead to errors.

**Pull request merged**